### ISSUE TITLE
The integration spams out manually start- and stop charging command for history dates

### ISSUE THREAD

**Issue by Sh1bbyz** [jonasbkarlsson/ev_smart_charging#44]
## Version of the custom_component
Version 1.0.4

## Configuration
Home Assistant 2022.11.5
Supervisor 2022.11.2
Operating System 9.3
Frontend 20221108.0 - latest

## Describe the bug
1) Smart charging sometimes does not start (did not start between the 2/12 and 3/12 for example)
Possibly because random commands to turn off "EV Smart Charging Charging" and point 2)
2) The integration spams out manually start- and stop charging command for history dates
3) The integration randomly spams out the command to turn off "EV Smart Charging Apply price limit" (altough it is already turned off)

## Debug log

**EV Smart charging event log:**

December 3, 2022

EV Smart Charging Manually stop charging changed to November 27, 2022 at 09:02
12:25:33 - 44 minutes ago
EV Smart Charging Manually start charging changed to November 28, 2022 at 21:28
12:25:33 - 44 minutes ago
EV Smart Charging Apply price limit turned off
12:25:32 - 44 minutes ago
EV Smart Charging Smart charging activated turned on
12:25:32 - 44 minutes ago
EV Smart Charging Charging turned off
12:25:32 - 44 minutes ago
December 2, 2022
EV Smart Charging Manually stop charging changed to November 27, 2022 at 09:02
19:57:18 - 17 hours ago
EV Smart Charging Manually start charging changed to November 28, 2022 at 21:28
19:57:18 - 17 hours ago
EV Smart Charging Charging turned off
19:57:18 - 17 hours ago
EV Smart Charging Manually stop charging changed to November 27, 2022 at 09:02
19:47:25 - 17 hours ago
EV Smart Charging Manually start charging changed to November 28, 2022 at 21:28
19:47:25 - 17 hours ago
EV Smart Charging Charging turned off
19:47:24 - 17 hours ago
EV Smart Charging Manually stop charging changed to November 27, 2022 at 09:02
19:08:13 - 18 hours ago
EV Smart Charging Manually start charging changed to November 28, 2022 at 21:28
19:08:13 - 18 hours ago
EV Smart Charging Apply price limit turned off
19:08:13 - 18 hours ago
EV Smart Charging Smart charging activated turned on
19:08:13 - 18 hours ago
EV Smart Charging Charging turned off
19:08:13 - 18 hours ago
EV Smart Charging Manually stop charging changed to November 27, 2022 at 09:02
15:27:27 - 22 hours ago
EV Smart Charging Manually start charging changed to November 28, 2022 at 21:28
15:27:27 - 22 hours ago
EV Smart Charging Charging turned off
15:27:27 - 22 hours ago
EV Smart Charging Manually stop charging changed to November 27, 2022 at 09:02
15:25:06 - 22 hours ago
EV Smart Charging Manually start charging changed to November 28, 2022 at 21:28
15:25:06 - 22 hours ago
EV Smart Charging Charging turned off
15:25:06 - 22 hours ago
EV Smart Charging Manually stop charging changed to November 27, 2022 at 09:02
15:23:11 - 22 hours ago
EV Smart Charging Manually start charging changed to November 28, 2022 at 21:28
15:23:11 - 22 hours ago
EV Smart Charging Charging turned off
15:23:11 - 22 hours ago

[HA_log.txt](https://github.com/jonasbkarlsson/ev_smart_charging/files/10146375/HA_log.txt)

**Comment by jonasbkarlsson** [comment]
Thanks for the report.

**Item 1 (charging not started)**

I noted in your HA_log.txt file the following
```
2022-12-03 12:25:32.787 ERROR (MainThread) [custom_components.ev_smart_charging.coordinator] SOC sensor not valid: <state sensor.zach_s_tazzzz_battery=unknown; state_class=measurement, raw_soc=None, unit_of_measurement=%, attribution=Data provided by Tesla, device_class=battery, icon=mdi:battery-unknown, friendly_name=Zach's Täzzzz Battery @ 2022-12-03T12:24:59.831653+01:00>
2022-12-03 12:25:32.811 ERROR (MainThread) [custom_components.ev_smart_charging.coordinator] Target SOC sensor not valid: <state number.zach_s_tazzzz_charge_limit=unknown; min=None, max=None, step=1, mode=auto, unit_of_measurement=%, attribution=Data provided by Tesla, icon=mdi:ev-station, friendly_name=Zach's Täzzzz Charge limit @ 2022-12-03T12:24:59.850362+01:00>

```
I also noted a lot of error messages from the custom component `tesla_custom`.

Is this "only" a startup problem, or do you have some issues with `tesla_custom`?

Can you check the values of `sensor.zach_s_tazzzz_battery` and `number.zach_s_tazzzz_charge_limit`? Do get a number between 0 and 100, or `unknown`?

**Item 2 (manual start- and stop charing commands in event log) and Item 3 (apply price limit commands in event log)**

This is the intended behavior when Home Assistant is restarted. Could it be so that your HA was restarted around 15:23, 15:25, 15:27, 19:08, 19:47 and 19:57 on December 2nd and around 12:25 on Decemember 3rd?

**Comment by Sh1bbyz** [comment]
Thanks for your reply and awsome integration! 
However the car did not charge tonight either :(

Item 1 (charging not started)
Hmm I see! It may have to do with that the values at times are unavailable on sensor.zach_s_tazzzz_battery and number.zach_s_tazzzz_charge_limit. This is because the car goes to sleep to keep the car from draining battery. That means that the smart charging cannot function?
I am wondering how to get around this and _when_. Is it only on startup the custom_components.ev_smart_charging.coordinator wants information from these or at certain times?

I am thinking about:
- Somehow add new sensors that saves the last known states (dunno how yet)
- Somehow add an helper that sends out "wake" command and starts polling the tesla. This requires to set the right conditions on when to do this and for how long.

Item 2 (manual start- and stop charing commands in event log) and Item 3 (apply price limit commands in event log)
Aha I see! But home come it sets the start- and stop dates to a history date? Same was done on reboot this morning, see below.

December 4, 2022
EV Smart Charging Manually stop charging changed to **November 27, 2022 at 09:02**
09:32:58 - 14 minutes ago
EV Smart Charging Manually start charging changed to **November 28, 2022 at 21:28**
09:32:58 - 14 minutes ago
EV Smart Charging Apply price limit turned off
09:32:58 - 14 minutes ago
EV Smart Charging Smart charging activated turned on
09:32:58 - 14 minutes ago
EV Smart Charging Charging turned off
09:32:58 - 14 minutes ago

**Comment by jonasbkarlsson** [comment]
**Item 1 (charging not started)**

The integration already use the last valid value of the SOC and Target_SOC entities, if any of them are unavailable. So adding a new sensor with the last known state shouldn't make a difference.

If the tesla sensors almost never updates, then that would be a problem, but is that really the case?

Another potential source or error is the Nordpool integration. There have been problems with it not updating tomorrow's price, and if that happens, there will not be any charging. Which version of the Nordpool integration are you using? Have you noticed any problems?

**Item 2 (manual start- and stop charing commands in event log) and Item 3 (apply price limit commands in event log)**

The EV Smart Charging Manually stop charging and EV Smart Charging Manually start charging are buttons. The state of the buttons that you see in the event log is the time when you last pushed them. That information is not used by the integration.

**Comment by Sh1bbyz** [comment]
Item 1 (charging not started)
Wierd that it didnt work then. The Tesla polls a couple of times a day. Perhaps it is still issue if you reset HA as the last known state also perhaps resets?

I use latest Nordpool integration and havent noticed any issues.

Anyway I have now been helped with the availability problem with the Tesla integration, thanks to your lead:
[https://github.com/alandtse/tesla/issues/311#issuecomment-1336441435](url)
I hope that this now solves the case and that the charging starts at 00.00 tonight. I will cheat charge a bit this evening though as the battery is totally drained and I really need to get to work tomorrow ;)

Item 2 (manual start- and stop charing commands in event log) and Item 3 (apply price limit commands in event log)
Alright, that explaines it! Really wierd for someone who doesnt know this fact though :)

**Comment by Sh1bbyz** [comment]
The car DID charge this night. It now seems to work! Thank you so much for the lead on what it could possibly be and thanks for the integration!

**Comment by jonasbkarlsson** [comment]
The integration does not remember the state of external entities across restarts. And without valid values for the SOC (and if configured, the Target SOC) entities, there will not be any charging. So if the integration that provides SOC and Target SOC does not provide valid values after restart, there will not be any charging.

This seems like being the case for you, right?

From this [comment ](https://github.com/alandtse/tesla/issues/311#issuecomment-1336441435)that you mentioned, in the case of the tesla integration, this can be avoided using the "Force cars awake on restart" option, right?

So can I close this issue?

**Comment by Sh1bbyz** [comment]
> The integration does not remember the state of external entities across restarts. And without valid values for the SOC (and if configured, the Target SOC) entities, there will not be any charging. So if the integration that provides SOC and Target SOC does not provide valid values after restart, there will not be any charging.
> 
> This seems like being the case for you, right?
> 
> From this [comment ](https://github.com/alandtse/tesla/issues/311#issuecomment-1336441435)that you mentioned, in the case of the tesla integration, this can be avoided using the "Force cars awake on restart" option, right?
> 
> So can I close this issue?

Yes I believe that this is the case and that the case now is closed!
